src/native.re

```diff
@@ -455,68 +481,60 @@ let finalOutputsScheme sortedSourcePaths::sortedSourcePaths => {
   Scheme.rules (List.append nativeRule javascriptRule)
 };
 
-/* The function that ties together all the previous steps and compiles a given library, whether it be our top
-   level library or a third-party one. */
 let compileLibScheme
-    isTopLevelLib::isTopLevelLib=true
-    srcDir::srcDir
+    libDir::libDir
+    libRoot::libRoot
     libName::libName
+    isTopLevelLib::isTopLevelLib
     buildDir::buildDir =>
-  Dep.glob_listing (Glob.create dir::srcDir "*.{re,rei,ml,mli}") |>
+  Dep.all (getSourceFiles dir::libDir |> List.map f::Dep.return) |>
```

Comment:
We need to make sure we can parallelize whatever we can by not over specifying our dependencies (jenga doesn't support specifying coarse dependencies unfortunately). Here we have no choice but to synchronously gather all the files for the module alias file, BUT compileSourcesScheme might be able to be parallelized.